release/platform/platform.go

```diff
@@ -0,0 +1,135 @@
+package platform
+
+import (
+	"errors"
+	"fmt"
+	"os"
+)
+
+// Platform represents a platform (a combination of OS, distro,
+// version, and architecture) on which we may build/test the tools.
+// There should be at least one evergreen buildvariant per platform,
+// and there may be multiple.
+type Platform struct {
+	Name string
+	Arch string
+}
+
+const evgPlatformVar = "EVG_PLATFORM"
+
+// Get returns the Platform for this host, based on the value of
+// EVG_PLATFORM. It returns an error if EVG_PLATFORM is unset or set
+// to an unknown value.
+func Get() (Platform, error) {
+	evgPlatform := os.Getenv(evgPlatformVar)
+	if evgPlatform == "" {
+		return Platform{}, errors.New("EVG_PLATFORM not set")
+	}
+
+	pf, ok := platforms[evgPlatform]
+	if !ok {
+		return Platform{}, fmt.Errorf("unknown evg platform id %q", evgPlatform)
+	}
+	return pf, nil
+}
+
+func IsWindows() (bool, error) {
+	p, err := Get()
+	if err != nil {
+		return false, err
+	}
+
+	switch p.Name {
+	case "windowsVS2017":
+		return true, nil
+	default:
+		return false, nil
+	}
+}
+
+var platforms = map[string]Platform{
+	"amazon1": {
+		Name: "amazon1",
+		Arch: "x86_64",
+	},
+	"amazon2": {
+		Name: "amazon2",
+		Arch: "x86_64",
+	},
+	"debian81": {
+		Name: "debian8",
+		Arch: "x86_64",
+	},
+	"debian92": {
+		Name: "debian9",
+		Arch: "x86_64",
+	},
+	"macos1014": {
+		Name: "macos1014",
+		Arch: "x86_64",
+	},
+	"rhel62": {
+		Name: "rhel62",
+		Arch: "x86_64",
+	},
+	"rhel70": {
+		Name: "rhel70",
+		Arch: "x86_64",
+	},
+	"suse12": {
+		Name: "suse12",
+		Arch: "x86_64",
+	},
+	"ubuntu1404": {
+		Name: "ubuntu1404",
+		Arch: "x86_64",
+	},
+	"ubuntu1604": {
+		Name: "ubuntu1604",
+		Arch: "x86_64",
+	},
+	"ubuntu1804": {
+		Name: "ubuntu1804",
+		Arch: "x86_64",
+	},
+	"windowsVS2017": {
+		Name: "windowsVS2017",
+		Arch: "x86_64",
+	},
+	"ubuntu1604-arm": {
+		Name: "ubuntu1604",
+		Arch: "arm",
```

Comment:
Server uses 'arm64'.

Reply:
updated

Reply:
I don't think many things use 32 bit arm anymore. Definitely nothing we want to run the tools on. It would likely be some really cheap embedded microcontroller (though a lot of those are actually 8 bit, still).